test/extended/builds/forcepull.go

```diff
@@ -1,183 +1,179 @@
+		serviceIP, err := oc.Run("get").Args("svc", "docker-registry", "-n", "default", "--config", exutil.KubeConfigPath()).Template("{{.spec.clusterIP}}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		port, err := oc.Run("get").Args("svc", "docker-registry", "-n", "default", "--config", exutil.KubeConfigPath()).Template("{{ $x := index .spec.ports 0}}{{$x.port}}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By(fmt.Sprintf("docker-registry service IP is %s and port %s ", serviceIP, port))
 
-				exutil.VerifyImagesSame(s2iDockBldr, custBldr, "FORCE PULL TEST s2i false:  ")
+		// get the auth so we can pull the build image from the internal docker registry since the builder controller will  remove it
+		// from the docker daemon cache when the docker build completes;
+		authCfg, err = exutil.BuildAuthConfiguration(serviceIP+":"+port, oc)
 
-			})
-		})
+		// now actually pull the image back in from the openshift internal docker registry
+		fullNameDS = authCfg.ServerAddress + "/" + oc.Namespace() + "/" + s2iDockBldr
+		err = exutil.PullImage(fullNameDS, *authCfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		fullNameC = authCfg.ServerAddress + "/" + oc.Namespace() + "/" + custBldr
+		err = exutil.PullImage(fullNameC, *authCfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
 
-		g.Context("\n FORCE PULL TEST:  when s2i force pull is true and the image is bad", func() {
-			g.It("\n FORCE PULL TEST s2i true", func() {
-				fpTrueS2I := exutil.FixturePath("fixtures", "forcepull-true-s2i.json")
-				g.By(fmt.Sprintf("\n%s FORCE PULL TEST s2i true:  calling create on %s", time.Now().Format(time.RFC850), fpTrueS2I))
-				exutil.StartBuild(fpTrueS2I, buildPrefix, oc)
+		//update the build configs in the json for the app/lang builds to point to the builder images in the internal docker registry
+		// and then create the build config resources
+		pre := exutil.FixturePath("fixtures", "forcepull-test.json")
+		post := exutil.ArtifactPath("forcepull-test.json")
+		varSubDest = authCfg.ServerAddress + "/" + oc.Namespace()
+		err = exutil.VarSubOnFile(pre, post, varSubSrc, varSubDest)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = exutil.CreateResource(post, oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
 
-				exutil.WaitForBuild("FORCE PULL TEST s2i true: ", buildName, oc)
+		// dump the image textual tags and hex ids out for debug
+		tags = []string{fullNameDS + ":latest", fullNameC + ":latest", corruptor + ":latest"}
+		hexIDs := exutil.DumpAndReturnTagging(tags)
+		resetData = map[string]string{fullNameDS: hexIDs[0], fullNameC: hexIDs[1], corruptor: hexIDs[2]}
 
-				exutil.VerifyImagesDifferent(s2iDockBldr, custBldr, "FORCE PULL TEST s2i true:  ")
-			})
-		})
+		// dump the image streams for debug compares later on
+		imageStreamOutput, err := oc.Run("get").Args("is", "-o", "json", "--config", exutil.KubeConfigPath()).Output()
+		g.By(fmt.Sprintf("image stream after builders are built - test run err %v and output %s ", err, imageStreamOutput))
```

Comment:
can we get this output to display only if the test failed?  does gingko automatically buffer all fmt.Print output and then only display it if the test fails?  if so that'd be the way i'd prefer to do this.